### DISCUSSION TITLE
Lighthouse snapshot feature in testcomplete

### DISCUSSION THREAD

**Discussion by mukeshmani4** [GoogleChrome/lighthouse#15463] [Q&A]
Recently im working on testcomplete and there is a requirement to incorporate accessibility audit , actually in test complete we have option to call lighthouse by web audit feature, but is there any way to call lighthouse snapshot feature so I can do a audit wherever required rather than calling lighthouse on the page load.

**Comment by adamraine** [comment]
Yes, you can use snapshot mode in a [user flow](https://github.com/GoogleChrome/lighthouse/blob/main/docs/user-flows.md) or use snapshot mode in the DevTools panel

**Reply by mukeshmani4** [reply]
Hey Adamraine thanks for your reply , but I'm looking to pass it as a function so it will be helpful if you provide me the  Java script code snippet or any relevant  article, what I'm trying here to accomplish is when the keyword test steps are executed by testcomplete I can call this audit function (snapshot mode ) wherever required.

**Reply by mukeshmani4** [reply]
@adamraine  so if I import start flow from lighthouse module , can I able to call snapshot . Will share my testcomplete code once I access the machine . Testcomplete is like record and play automation tool so my goal is to record the steps / navigation using testcomplete and call the audit function using javascript

**Reply by adamraine** [reply]
You cannot call Lighthouse code from within the browser tab that you are testing.

Lighthouse uses https://github.com/dequelabs/axe-core for it's accessibility advice. You might be able to get something working with just axe-core code to test your page that way.
